src/testcases/org/apache/poi/ss/formula/TestWorkbookEvaluator.java

```diff
@@ -606,4 +608,38 @@ private void testIFEqualsFormulaEvaluation_evaluateAllFormulaCells(
         
         testIFEqualsFormulaEvaluation_teardown(wb);
     }
+
+    @Test
+    public void testSheetNameReference() throws IOException {
+        Workbook wb = new HSSFWorkbook();
+
+        Sheet s1 = wb.createSheet("Sheet1");
+        Cell cell1 = CellUtil.getCell(CellUtil.getRow(0, s1), 0);
+        cell1.setCellValue(123);
+        Sheet s2 = wb.createSheet();
+        Cell cell2 = CellUtil.getCell(CellUtil.getRow(0, s2), 0);
+
+        FormulaEvaluator eval = wb.getCreationHelper().createFormulaEvaluator();
+        {
+            cell2.setCellFormula("Sheet1!A1");
+            CellValue value = eval.evaluate(cell2);
+            assertEquals(123d, value.getNumberValue(), 0);
+        }
+
+        for (int i = 256; i <= 65535; i++) {
```

Comment:
To make this unit test run faster, could you pick just a few exemplary characters that should run test whether POI behaves correctly or not? For example, one character from A-Z, one character from a-z (to test case sensitivity), one character from 0-9, one letter from unicode set (ß), one non-whitespace character from unicode set (halfwidth ¥ and fullwidth ￥), several ascii whitespace (space, \n, \r, \t), a unicode whitespace (en space).

    for (char c : new char[] { ... })